File: src/main/java/org/example/chessplatformbe/controller/dto/request/CreateStreamDTO.java
package org.example.chessplatformbe.controller.dto.request;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CreateStreamDTO {
    private Integer updateId;

    @NotBlank(message = "Name is required")
    private String name;

    private LocalDateTime creationDateTime;

    @NotBlank(message = "Stream Url is required")
    private String streamUrl;

    @NotNull(message = "Streamer id is required")
    private Integer streamerId;
}

File: src/controller/dto/request/create_stream_dto.rs
use chrono::NaiveDate;

pub struct CreateStreamDTO {
    update_id: u32,
    name: String,
    creation_date_time: NaiveDate,
    stream_url: String,
    streamer_id: u32,
}

impl CreateStreamDTO {
    pub fn new(
        update_id: u32,
        name: String,
        creation_date_time: NaiveDate,
        stream_url: String,
        streamer_id: u32,
    ) -> Self {
        Self {
            update_id,
            name,
            creation_date_time,
            stream_url,
            streamer_id,
        }
    }
}
